docs/proposals/cwag_ha/cwag_ha_transport_failover.md

```diff
@@ -0,0 +1,104 @@
+# CWAG HA Transport Failover - Design Proposal
+
+*Status: In-Review*\
+*Author: @mpgermano*\
+*Reviewers: @cwf-team*\
+*Last Updated: 07/24*
+
+# Overview
+
+In H1 of 2020, Magma implemented CWAG high availability through an active/standby design. 
+The initial design relied on the external WLC/AP to properly route RADIUS and GRE traffic 
+to the IP of the active gateway. This document outlines a proposal to handle transport 
+failover as a part of the CWAG HA failover mechanism.
+
+## Transport Failover
+
+The HA design currently relies on the WLC/AP to support redundant GRE tunnels and RADIUS servers. 
+It works in the following way:
+
+1. When the active becomes unhealthy, an ICMP drop rule is added to eth1 of the failed active. 
+   The RADIUS server is also stopped for this gateway.
+2. The WLC detects that the primary transport (both GRE and RADIUS) is down.
+3. After a configurable number of seconds, the WLC will begin using the secondary server 
+   and tunnel configured. 
+
+This design is problematic for a couple of reasons:
+
+* Not all WLC/APs support transport redundancy (e.g. Cambium)
+* The failover determination is separate for the RADIUS and GRE transports. This means that
+  control-plane traffic could be sent to the standby while data-plane traffic is routed to 
+  the active (or vice versa).
+
+To fix this, we want to have a single IP that can properly route traffic to whichever gateway
+is currently designated as active. 
+
+### Using a Virtual IP
+
+We can configure a virtual IP address in the NMS as a network level config that will be 
+streamed to each CWAG in the cluster. This VIP will use device eth1 of the CWAG. For example:
+
+```
+eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST> mtu 1500
+ inet 10.22.20.4 netmask 255.255.255.0 broadcast 10.22.20.255
+ ether e6:a1:54:03:c0:2f txqueuelen 1000 (Ethernet)
+ 
+eth1:0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST> mtu 1500
+ inet **10.22.20.5** netmask 255.255.255.0 broadcast 10.255.255.255
+ ether e6:a1:54:03:c0:2f txqueuelen 1000 (Ethernet)
+```
+
+The CWF Operator, which manages the active/standby status and associated initialization, will be 
+responsible for triggering the transfer of the IPs on failover. When a failover occurs, the standby 
+is initialized first. If that succeeds, the active is then initialized. The happy path for a 
+failover will consist of the following steps:
+
+Standby:
+* Delete the virtual IP 
+
+Active:
+* Add the virtual IP
+* Send gratuitous ARP for the VIP
```

Comment:
You should send a reply arp I think

Reply:
Yeah this should say `gratuitous ARP Reply`